EX05/client/app/src/main/java/com/example/stockmonitor/MyFirebaseMessagingService.java

```diff
@@ -0,0 +1,104 @@
+package com.example.stockmonitor;
+
+import android.app.Service;
+import android.content.Intent;
+import android.os.IBinder;
+import android.util.Log;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.Volley;
+import com.google.firebase.messaging.FirebaseMessagingService;
+import com.google.firebase.messaging.RemoteMessage;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Map;
+
+public class MyFirebaseMessagingService extends FirebaseMessagingService {
+
+    private final static String REQUEST_URL = "http://10.0.2.2:8080/initToken";
```

Comment:
Better to get from configuration or something so you don't have to re-compile your code if you want to change the REQUEST_URL.